buildpacks/nodejs-function-invoker/src/main.rs

```diff
@@ -0,0 +1,140 @@
+        is_function(&context.app_dir)
+            .then(|| {
+                DetectResultBuilder::pass()
+                    .build_plan(
+                        BuildPlanBuilder::new()
+                            .requires("node")
+                            .requires("nodejs-function-invoker")
+                            .provides("nodejs-function-invoker")
+                            .build(),
+                    )
+                    .build()
+            })
+            .unwrap_or_else(|| DetectResultBuilder::fail().build())
+    }
+
+    fn build(&self, context: BuildContext<Self>) -> libcnb::Result<BuildResult, Self::Error> {
+        log_header("Heroku Node.js Function Invoker Buildpack");
+
+        log_info("Checking for function file");
+        get_main(&context.app_dir).map_err(NodeJsInvokerBuildpackError::MainFunctionError)?;
+
+        context.handle_layer(
+            layer_name!("runtime"),
+            RuntimeLayer {
+                package: context
+                    .buildpack_descriptor
+                    .metadata
+                    .runtime
+                    .package
+                    .clone(),
+            },
+        )?;
+
+        BuildResultBuilder::new()
+            .launch(
+                Launch::new().process(
+                    ProcessBuilder::new(process_type!("web"), "sf-fx-runtime-nodejs")
+                        .args(vec![
+                            "serve",
+                            &context.app_dir.to_string_lossy(),
+                            "--workers",
+                            "2",
+                            "--host",
+                            "::",
+                            "--port",
+                            "${PORT:-8080}",
+                            "--debug-port",
+                            "${DEBUG_PORT:-}",
+                        ])
+                        .default(true)
+                        .build(),
+                ),
+            )
+            .build()
+    }
+
+    fn on_error(&self, error: libcnb::Error<Self::Error>) -> i32 {
+        match error {
+            libcnb::Error::BuildpackError(bp_err) => {
+                let err_string = bp_err.to_string();
+                match bp_err {
+                    NodeJsInvokerBuildpackError::MainFunctionError(_) => {
+                        log_error(
+                            "Node.js Function Invoker main function detection error",
+                            err_string,
+                        );
+                        70
+                    }
+                    NodeJsInvokerBuildpackError::RuntimeLayerError(_) => {
+                        log_error("Node.js Function Invoker runtime layer error", err_string);
+                        71
+                    }
+                }
+            }
+            err => {
+                log_error("Internal Buildpack Error", err.to_string());
+                100
+            }
+        }
+    }
+}
```

Comment:
This could leverage `libherokubuildpack::on_error_heroku` to simplify error handling a little bit and benefit from cross-language error handling that will be eventually added to `libherokubuildpack`.

Reply:
I didn't use `on_error_heroku` in `heroku/nodejs_engine` mostly because I didn't see the value. I still don't see much value, but I guess it might be there one day. I don't feel very strongly, so it's fixed in https://github.com/heroku/buildpacks-nodejs/pull/224/commits/8040d4177082b4da311f033245da498fd7301dfe.